spec/ruby/core/module/const_added_spec.rb

```diff
@@ -121,5 +121,22 @@ class SubClass
 
       ScratchPad.recorded.should == [line + 2, line + 4, line + 7, line + 11]
     end
+
+    it "is called when the constant is ready to be used" do
+      ScratchPad.record []
+
+      mod = Module.new do
+        def self.const_added(name)
+          ScratchPad << const_get(name)
+        end
+      end
+
+      mod.module_eval(<<-RUBY, __FILE__, __LINE__ + 1)
+        TEST = 123
+        TEST = 456
+      RUBY
+
+      ScratchPad.recorded.should == [123, 456]
+    end
```

Comment:
**suggestion:** It seems the callback is called every time a constant is re-assigned a new value. It makes sense to have a separate test case to illustrate this behaviour.

Reply:
Sure thing